protocol/messenger.go

```diff
@@ -2461,6 +2461,10 @@ func (m *Messenger) SyncDevices(ctx context.Context, ensName, photoPath string,
 }
 
 func (m *Messenger) SaveAccounts(accs []*accounts.Account) error {
+	clock, _ := m.getLastClockWithRelatedChat()
+	for _, acc := range accs {
```

Comment:
Unfortunately this might not be correct, say that you updated a single field on a single account, then each account will have their clock modified, and they will overwrite potential changes on other accounts.

Basically this assumes that Accounts (all of them) are a single entity and synchronize together, which is a bit risky. 

How do we update accounts, do you have a link to desktop?

Reply:
The only place where this `(m *Messenger) SaveAccounts` function is called from is from the `(api *API) SaveAccounts` that further means it is used by client only.

Even `SaveAccounts` receives an array of accounts (which dates from who knows when and who knows why) that array in 100% cases has only a single element. User is not able to make changes over 2 or more accounts simultaneously. Even if that's possible that also means all those accounts are changed at the same time, so all of them need to have the same `Clock` in that case.

> Basically this assumes that Accounts (all of them) are a single entity and synchronize together, which is a bit risky.

This part doesn't have anything with the synchronization or backing up accounts. Also handling accounts received from synchronization or backed up messages do not use this function. It's only used when client interacts with an account(s) and sets the clock which is set to db and used to notify other paired devices when the change was made over that account and used at the moment of backing up accounts.

> How do we update accounts, do you have a link to desktop?

The desktop app updates each individual account separately, also the desktop app doesn't set any Clock value either it's a new account creation or an account update action.

Reply:
> The only place where this (m *Messenger) SaveAccounts function is called from is from the (api *API) SaveAccounts that further means it is used by client only.

Ok, so it's not saving all the existing accounts at the same time, only those that have changed, makes sense.

> This part doesn't have anything with the synchronization or backing up accounts

Maybe we misunderstand each other, but clock is exclusively used for synchronization/backup when it comes to wallet, and we sync just a few lines below, so looks like is very much related to synchronization this code.
But my concern has been addressed, I assumed `SaveAccounts` will always to take the full account list (changed/unchanged), but if it's only used for changed accounts, the code above is correct.

Reply:
Yes, we're always storing only changed field/s for a single account when the change is made. Finally on the same page 🙂

Reply:
>  we're always storing only changed field/s for a single account 

emm.. it would be better to change `func (m *Messenger) SaveAccounts(accs []*accounts.Account) error` to `func (m *Messenger) SaveAccount(acc *accounts.Account) error` although we checked we're always storing only single account. We can't gurantuee future developer won't make mistake 🙂 @saledjenic

Reply:
@qfrank I updated the `SaveAccounts` endpoint, please chek [this commit](https://github.com/status-im/status-go/pull/3416/commits/07575eebd1999d5defb82fb29a26b730005009af).